Added exporting cpu/mem limits based on Kubernetes rather than cadvisor

piosz opened the pull request:
Fixes #399

piosz commented:
@vmarmol

#399 was only about fixing cpu, but I've also fixed memory to make both resources exported consistently.

vmarmol commented on sources/nodes/types.go:
```diff
@@ -30,6 +30,10 @@ type Info struct {
 	// An optional ID assigned to nodes by either the cloud provider or user.
 	ExternalID string
+	// Cpu capacity of node as number of 1/1024 share of core.
+	CpuCapacity uint64
```
This is exporting in shares? At the Heapster level (and for nodes) we should probably export in whole cores. Shares is hard to reason about at a node level.

piosz replied:
Just wanted to make it consistent with the current unit, but it will be definitely easier to use millicores instead.

vmarmol commented:
Outside the issue with shares vs. cores, LGTM

piosz commented:
Changed to use millicores internally. We probably don't want to use whole core (not millicores) to support also machines with fractional part of virtual CPU. Please let me know if it addressed your comment.

vmarmol commented on sinks/api/v1/supported_metrics.go:
```diff
@@ -66,8 +66,7 @@ var statMetrics = []SupportedStatMetric{
 		GetValue: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) []InternalPoint {
-			// Normalize to a conversion factor of 1000.
-			return []InternalPoint{{Value: int64(spec.Cpu.Limit*1000) / 1024}}
+			return []InternalPoint{{Value: int64(spec.Cpu.Limit)}}
```
I would have expected we still do this? We fixed it for node containers but not for general containers no?

piosz replied:
I see the problem. Fixed.

piosz commented:
ping

vmarmol commented:
LGTM, sorry this dropped out of my radar!

piosz commented:
No worries. Thanks you!
